### PR TITLE
Icon component

### PR DESCRIPTION
**Pull request by saksham695**

### CODE REVIEW

File: examples/App.js [hunk 1]
```diff
@@ -11,18 +11,38 @@
...
 import React from 'react';
 import UiBuilder from './src/index';
-import { View } from 'react-native';
+import {View} from 'react-native';
 
 const onButtonClick = () => {
   alert('Button Pressed');
 };
+const onIconClick = () => {
+  alert('Icon Pressed');
```
Comment (mahima-vs) [inline examples/App.js:20]: You can give a default message in default props, message can be like- "Please add message on icon click".

File: examples/src/components/icon.js [hunk 2]
```diff
@@ -1,24 +1,34 @@
...
 import React, {Component} from 'react';
-import {View, StyleSheet, Text} from 'react-native';
 import {Icon} from 'react-native-elements';
-import PropTypes from 'prop-types';
-import {theme} from '../../App';
+import {View, StyleSheet, Text} from 'react-native';
 import {applyThemeOnIconStyle} from '../theme/iconTheme';
+import {theme} from '../../App';
+
+// TODO :remove warnings laters
```
Comment (mahima-vs) [inline examples/src/components/icon.js:8]: Please remove warnings before merging.

File: examples/src/components/icon.js [hunk 3]
```diff
@@ -1,24 +1,34 @@
...
+  }
+
   render() {
-    //  console.log(this.props);
+    const style = {
+      marginRight: 40,
+      alignSelf: 'flex-end',
+      marginTop: 190,
+    };
     props = this.props;
```
Comment (mahima-vs) [inline examples/src/components/icon.js:20]: Where are you declare "props"?

File: examples/src/theme/iconTheme.js [hunk 4]
```diff
@@ -3,21 +3,20 @@ import {getThemeStyle} from '../utility/utils';
 // function to change old style without theme to stylw with theme
```
Comment (mahima-vs) [inline examples/src/theme/iconTheme.js:3]: typo, please check spellings.

File: examples/src/theme/iconTheme.js [hunk 5]
```diff
@@ -3,21 +3,20 @@ import {getThemeStyle} from '../utility/utils';
...
 
 export const applyThemeOnIconStyle = (styleWithouthTheme, theme) => {
```
Comment (mahima-vs) [inline examples/src/theme/iconTheme.js:5]: Write algorithm or a short description about the method.

File: examples/src/components/icon.js (hunk 3 above)
Comment (saksham695) [reply examples/src/components/icon.js:20]: props are declare in line 20 , line 15-19 was used to just check styling . I will remove them , they aren't used anywhere in component .

Comment (mahima-vs) [reply examples/src/components/icon.js:20]: At line 20, you are doing the assignment, declaration means const props.

File: examples/src/theme/iconTheme.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import {getThemeStyle} from '../utility/utils';
+
+// function to change old style without theme to style with theme
```
Comment (mahima-vs) [inline examples/src/theme/iconTheme.js:3]: Comment should be- Method to apply a theme on the icon.

File: examples/src/theme/iconTheme.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+export const applyThemeOnIconStyle = (styleWithouthTheme, theme) => {
```
Comment (mahima-vs) [inline examples/src/theme/iconTheme.js:5]: styleWithouthTheme, typo

File: examples/App.js (hunk 1 above)
Comment (saksham695) [reply examples/App.js]: done

File: examples/src/components/icon.js (hunk 2 above)
Comment (saksham695) [reply examples/src/components/icon.js]: done

File: examples/src/components/icon.js (hunk 3 above)
Comment (saksham695) [reply examples/src/components/icon.js]: done

File: examples/src/theme/iconTheme.js (hunk 4 above)
Comment (saksham695) [reply examples/src/theme/iconTheme.js]: done

File: examples/src/theme/iconTheme.js (hunk 5 above)
Comment (saksham695) [reply examples/src/theme/iconTheme.js:5]: done

File: examples/src/theme/iconTheme.js (hunk 6 above)
Comment (saksham695) [reply examples/src/theme/iconTheme.js:3]: done

File: examples/src/theme/iconTheme.js (hunk 7 above)
Comment (saksham695) [reply examples/src/theme/iconTheme.js:5]: done

File: examples/src/components/icon.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+import {View, StyleSheet, Text} from 'react-native';
+import {theme} from '../../App';
+import {applyTheme} from '../utility/utils';
+export default class IconComponent extends Component {
+  constructor() {
+    super();
+  }
+
+  render() {
+    this.props = theme ? applyTheme(this.props, theme) : this.props;
```
Comment (mahima-vs) [inline examples/src/components/icon.js:13]: Identify a method from a react lifecycle that will call before render and put this piece in that method.

File: examples/src/utility/utils.js [hunk 9]
```diff
@@ -1,27 +1,52 @@
...
+  placeholderTextColor: 'white',
+  borderColor: 'white',
+};
+
+// method to apply theme
+export const applyTheme = (componentStyle, theme) => {
+  const existingStyle = {...componentStyle};
+  const themeStyle = getThemeStyle(theme);
+  // get key values from theme
+  const keys = Object.keys(themeStyle);
```
Comment (mahima-vs) [inline examples/src/utility/utils.js:32]: No need of making a const for "keys", use Object.keys directly wherever reqd.

**Pull request merged**